### ISSUE TITLE
provider/aws: aws_vpc_peering_connection requires peer_owner_id

### ISSUE THREAD

**Issue by artburkart** [hashicorp/terraform#3995]
I know the Terraform docs [say it's required](https://terraform.io/docs/providers/aws/r/vpc_peering.html#peer_owner_id), but the AWS docs say it is not because it [defaults to your AWS account id](http://docs.aws.amazon.com/cli/latest/reference/ec2/create-vpc-peering-connection.html#options). If I changed the code that forces it as required, would it work by default? 

Of course, I know I can just set the value. This is more of a question of interest and the level of dev effort to bring the terraform api a bit closer to the AWS api.

**Comment by jen20** [comment]
@artburkart You should just be able to change the `Required: true` to `Optional: true` in the schema, and update the documentation if you want to take a look at this. I'll mark this as a bug and someone will look in due course, but if you feel like taking it on then that's great!

**Comment by artburkart** [comment]
I'll see if I can take a look this weekend, thanks!

**Comment by artburkart** [comment]
@jen20 - I switched the parameter for `resourceAwsVpcPeeringConnection` to `Optional: true` and I get this error message:

```console
* aws_vpc_peering_connection.peer_connection: Error creating vpc peering connection: MissingParameter: The request must contain the parameter peerOwnerId
	status code: 400, request id:
```

This error means it's a bug in the AWS Go SDK, right? I was going to attempt to create a `DefaultFunc` for the `resourceAwsVpcPeeringConnection`, but I wasn't sure it made sense, since the original author of the code probably only set `peer_owner_id` to required because the AWS Go SDK didn't function properly.

**Comment by pll** [comment]
Has this been fixed in recent releases?  Is peer_owner_id still required by terraform?

**Comment by fazy** [comment]
Using 0.6.12 it seems definitely required still (using example from doc but removing the `peer_owner_id` line):

```
Errors:

  * aws_vpc_peering_connection.foo: "peer_owner_id": required field is not set
```

**Comment by nocode99** [comment]
FYI - This was fixed in `0.7.4` https://github.com/hashicorp/terraform/issues/6396

**Comment by radeksimko** [comment]
Thanks for the ping @nocode99 
This was indeed fixed in https://github.com/hashicorp/terraform/commit/38d2a2e71770b480187a6e50df8af2472637c957 (part of `0.7.3+`).
